### PR TITLE
Add cmake build for Windows

### PR DESCRIPTION
**Pull request by matthewbastien**
This is tracked by rdar://139431575 as a part of https://github.com/swiftlang/vscode-swift/issues/562

Add a cmake build of swift-lmdb for Windows that can be used by [swiftlang/indexstore-db](https://github.com/swiftlang/indexstore-db) to match the version of LMDB used by swift-docc. This enables SourceKit-LSP to use swift-docc as a dependency to fulfill documentation requests on behalf of the VS Code Swift extension.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.24)
+
+project(LMDB
+  LANGUAGES C)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
```
Comment (compnerd) [inline CMakeLists.txt]: Is there Swift code in the project? I think that L10 can be dropped.

File: Sources/CLMDB/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+add_library(CLMDB STATIC
```
Comment (compnerd) [inline Sources/CLMDB/CMakeLists.txt]: Why explicitly static? Why not drop `STATIC` and let the user control the library type?

**Comment by compnerd** [comment]
BTW, what is the difference between this and #6? I can unmark that as a draft if that is helpful.

**Comment by matthewbastien** [comment]
> BTW, what is the difference between this and #6? I can unmark that as a draft if that is helpful.

I don't think there is a difference at all, really. If that PR is ready then I'm fine with using it instead. I'm really not all that familiar with cmake. So, I'm mostly doing this by trial and error and looking at other projects.

**Comment by compnerd** [comment]
@matthewbastien yes, that PR has been ready, the only reason it was marked as a draft is because I didn't have the DocC change ready to go yet (I have a CMake based build of DocC which was significantly reducing the build times and somewhat on the size as well).

**Comment by matthewbastien** [comment]
Awesome, thank you! I'm closing this PR in favour of yours.

**Pull request closed without merging**